core/trino-main/src/main/java/io/trino/server/NodeStateManager.java

```diff
@@ -69,61 +75,114 @@ public NodeStateManager(
         this.gracePeriod = serverConfig.getGracePeriod();
     }
 
-    public synchronized void requestShutdown()
+    public NodeState getServerState()
     {
-        log.info("Shutdown requested");
+        return nodeState.get();
+    }
 
-        if (isCoordinator) {
-            throw new UnsupportedOperationException("Cannot shutdown coordinator");
-        }
+    /*
+    Below is a diagram with possible states and transitions
+
+    @startuml
+    [*] --> ACTIVE
+    note "state INACTIVE is not used internally\nis only used when the service is unavailable " as a
```

Comment:
what is this note about. There is no `INACTIVE` state here

Reply:
Oh - you refer to enum value

Reply:
yes, do you want me to clarify this comment?